src/librustc/traits/select.rs

```diff
@@ -2086,12 +2086,8 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
 
             ty::TyClosure(def_id, substs) => {
                 let trait_id = obligation.predicate.def_id();
-                let copy_closures =
-                    Some(trait_id) == self.tcx().lang_items().copy_trait() &&
-                    self.tcx().has_copy_closures(def_id.krate);
-                let clone_closures =
-                    Some(trait_id) == self.tcx().lang_items().clone_trait() &&
-                    self.tcx().has_clone_closures(def_id.krate);
+                let copy_closures = Some(trait_id) == self.tcx().lang_items().copy_trait();
+                let clone_closures = Some(trait_id) == self.tcx().lang_items().clone_trait();
```

Comment:
I’m not confident about this particular chunk, but tests appear to be happy.

Reply:
seems ok. The name of the variable is weird; I would prefer `is_copy_trait` and `is_clone_trait` or something.